docker/docker_client.go

```diff
@@ -248,33 +247,30 @@ func newDockerClient(sys *types.SystemContext, registry, reference string) (*doc
 	// Check if TLS verification shall be skipped (default=false) which can
 	// either be specified in the sysregistriesv2 configuration or via the
 	// SystemContext, whereas the SystemContext is prioritized.
-	skipVerify := false
+	var sysContextSkipVerify *bool
 	if sys != nil && sys.DockerInsecureSkipTLSVerify != types.OptionalBoolUndefined {
 		// Only use the SystemContext if the actual value is defined.
-		skipVerify = sys.DockerInsecureSkipTLSVerify == types.OptionalBoolTrue
-	} else {
-		reg, err := sysregistriesv2.FindRegistry(sys, reference)
```

Comment:
This needs to continue to exist for non-`newImageSource` callers, _somehow_.

Maybe `newImageSource` should pass the right value (into this function? Not immediately sure. (`insecureSkipTLSVerify *bool`?)

Reply:
I decided to readd the code path again as well as the `reference` variable. Now `endpointSkipVerify` will be set by `newDockerClient`, too (but it can be manually set afterwards as well).